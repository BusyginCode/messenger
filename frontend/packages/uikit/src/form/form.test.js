import React from 'react'
import { shallow, mount } from 'enzyme'
import Form from './form'
import Button from '../button'

describe('uikit/Input', () => {
  it('should renders correctly', () => {
    const form = shallow(<Form />)
    expect(form).toMatchSnapshot()
  })

  it('should handle submit', () => {
    const handleSubmit = jest.fn()
    const form = mount(
      <Form onSubmit={handleSubmit}>
        <Button type="submit">submit</Button>
      </Form>
    )
    form.find('button').simulate('submit')
    expect(handleSubmit).toHaveBeenCalledTimes(1)
  })

  describe('#handleSubmit', () => {
    let handleSubmit
    beforeAll(() => {
      const form = shallow(<Form />)
      handleSubmit = form.instance().handleSubmit
    })

    it('should prevent default event on submit', () => {
      const preventDefault = jest.fn()
      handleSubmit({ preventDefault })
      expect(preventDefault).toHaveBeenCalledTimes(1)
    })
  })
})
