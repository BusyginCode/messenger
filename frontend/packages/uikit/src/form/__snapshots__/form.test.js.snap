// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uikit/Input should renders correctly 1`] = `
<form
  onSubmit={[Function]}
/>
`;
